modules/data.atmosphere/R/download.ICOS.R

```diff
@@ -0,0 +1,233 @@
+#' Download ICOS Ecosystem data products
+#'
+#' Currently available products: 
+#' Drought-2018 ecosystem eddy covariance flux product https://www.icos-cp.eu/data-products/YVR0-4898
+#' ICOS Final Fully Quality Controlled Observational Data (Level 2) https://www.icos-cp.eu/data-products/ecosystem-release
+#'
+#'
+#' @param sitename ICOS id of the site. Example - "BE-Bra"
+#' @param outfolder path to the directory where the output file is stored. If specified directory does not exists, it is created.
+#' @param start_date start date of the data request in the form YYYY-MM-DD
+#' @param end_date end date area of the data request in the form YYYY-MM-DD
+#' @param product ICOS product to be downloaded. Currently supported options: "Drought2018", "ETC"
+#' @param overwrite should existing files be overwritten. Default False.
+#' @return information about the output file
+#' @export
+#' @examples
+#' \dontrun{
+#' download.ICOS("FI-Sii", "/home/carya/pecan", "2016-01-01", "2018-01-01", product="Drought2018")
+#' }
+#' @author Ayush Prasad
+#'
+download.ICOS <-
+  function(sitename,
+           outfolder,
+           start_date,
+           end_date,
+           product,
+           overwrite = FALSE) {
+    download_file_flag <- TRUE
+    extract_file_flag <- TRUE
+    
+    
+    if (tolower(product) == "drought2018") {
+      # construct output CSV file name
+      output_file_name <-
+        paste0(
+          "FLX_",
+          sitename,
+          "_FLUXNET2015_FULLSET_HH_",
+          as.character(format(as.Date(start_date), '%Y')),
+          "-2018_beta-3.csv"
+        )
+      
+      # construct zip file name
+      zip_file_name <-
+        paste0(outfolder, "/Drought", sitename, ".zip")
+      
+      # data type, can be found from the machine readable page of the product
+      data_type <-
+        "http://meta.icos-cp.eu/resources/cpmeta/dought2018ArchiveProduct"
+      
+      file_name <-
+        paste0('FLX_', sitename, '_FLUXNET2015_FULLSET_HH')
+      
+      format_name <- "ICOS_ECOSYSTEM_HH"
+      
+    } else if (tolower(product) == "etc") {
+      output_file_name <-
+        paste0(sitename, "/ICOSETC_", sitename, "_FLUXNET_HH_01.csv")
```

Comment:
can we get rid of this `sitename` subfolder?
```suggestion
        paste0("ICOSETC_", sitename, "_FLUXNET_HH_01.csv")
```

Reply:
yes, thanks I wasn't aware of this `junkpaths`